src/bin/wallet.rs

```diff
@@ -1,39 +1,50 @@
 #[macro_use]
 extern crate clap;
 extern crate dirs;
+#[macro_use]
 extern crate solana;
 
 use clap::{App, Arg, ArgMatches, SubCommand};
 use solana::client::mk_client;
-use solana::crdt::NodeInfo;
 use solana::drone::DRONE_PORT;
 use solana::logger;
 use solana::signature::{read_keypair, KeypairUtil};
 use solana::thin_client::poll_gossip_for_leader;
-use solana::wallet::{parse_command, process_command, read_leader, WalletConfig, WalletError};
+use solana::wallet::{parse_command, process_command, WalletConfig, WalletError};
 use std::error;
-use std::net::{IpAddr, Ipv4Addr, SocketAddr};
+use std::net::SocketAddr;
+use std::process::Command;
 
 pub fn parse_args(matches: &ArgMatches) -> Result<WalletConfig, Box<error::Error>> {
-    let leader: NodeInfo;
-    if let Some(l) = matches.value_of("leader") {
-        leader = read_leader(l)?.node_info;
+    let network = if let Some(addr) = matches.value_of("network") {
+        addr.parse().or_else(|_| {
+            Err(WalletError::BadParameter(
+                "Invalid network location".to_string(),
+            ))
+        })?
     } else {
-        let server_addr = SocketAddr::new(IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)), 8000);
-        leader = NodeInfo::new_with_socketaddr(&server_addr);
+        socketaddr!("127.0.0.1:8001")
     };
-    let timeout: Option<u64>;
-    if let Some(secs) = matches.value_of("timeout") {
-        timeout = Some(secs.to_string().parse().expect("integer"));
+    let timeout = if let Some(secs) = matches.value_of("timeout") {
+        Some(secs.to_string().parse().expect("integer"))
     } else {
-        timeout = None;
-    }
+        None
+    };
 
     let mut path = dirs::home_dir().expect("home directory");
     let id_path = if matches.is_present("keypair") {
         matches.value_of("keypair").unwrap()
     } else {
         path.extend(&[".config", "solana", "id.json"]);
+        if !path.exists() || matches.is_present("reset") {
+            Command::new("cargo")
```

Comment:
Let's not assume `cargo` is available in the system.  Instead we can import the relevant solana-keygen code and invoke it here as a function.